src/main/java/net/javadiscord/javabot/systems/notification/GuildNotificationService.java

```diff
@@ -1,62 +1,50 @@
 package net.javadiscord.javabot.systems.notification;
 
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import net.dv8tion.jda.api.entities.Guild;
-import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.MessageChannel;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
 import net.javadiscord.javabot.Bot;
-import net.javadiscord.javabot.data.config.GuildConfig;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Function;
 
 /**
- * Sends notifications within a single {@link Guild}.
+ * Handles all sorts of guild notifications.
  */
 @Slf4j
-public final class GuildNotificationService extends NotificationService {
+@RequiredArgsConstructor(access = AccessLevel.PACKAGE)
+public final class GuildNotificationService extends NotificationService.MessageChannelNotification {
```

Comment:
Maybe make this class package private?

Reply:
Why? This wouldn't let me access the classes' methods from the outside

Reply:
I forgot to delete that comment after looking at the other classes.